CMakeLists.txt

```diff
@@ -8,20 +8,40 @@ PROJECT(HPXMP CXX ASM)
 
 cmake_minimum_required(VERSION 2.8)
 
-set(hpxmp_sources
-        src/asm_functions.s
-        src/intel_hpxMP.cpp
-        src/hpx_runtime.cpp
-        src/kmp_atomic.cpp
-        src/loop_schedule.cpp
-        src/gcc_hpxMP.cpp)
-
-set(hpxmp_headers
-        src/intel_hpxMP.h
-        src/hpx_runtime.h
-        src/kmp_atomic.h
-        src/loop_schedule.h
-        src/gcc_hpxMP.h)
+execute_process(COMMAND gcc -dumpmachine OUTPUT_STRIP_TRAILING_WHITESPACE OUTPUT_VARIABLE ARCH)
+
+if(ARCH STREQUAL "riscv64-linux-gnu")
+    set(hpxmp_sources
+            src/riscv64_asm_functions.s
+            src/intel_hpxMP.cpp
+            src/hpx_runtime.cpp
+            src/kmp_atomic.cpp
+            src/loop_schedule.cpp
+            src/gcc_hpxMP.cpp)
+
+    set(hpxmp_headers
+            src/intel_hpxMP.h
+            src/hpx_runtime.h
+            src/kmp_atomic.h
+            src/loop_schedule.h
+            src/gcc_hpxMP.h)
+
+else()
+    set(hpxmp_sources
+            src/asm_functions.s
+            src/intel_hpxMP.cpp
+            src/hpx_runtime.cpp
+            src/kmp_atomic.cpp
+            src/loop_schedule.cpp
+            src/gcc_hpxMP.cpp)
+
+    set(hpxmp_headers
+            src/intel_hpxMP.h
+            src/hpx_runtime.h
+            src/kmp_atomic.h
+            src/loop_schedule.h
+            src/gcc_hpxMP.h)
+endif()
```

Comment:
Could you please factor out the files that are used always and have only the platform specific files listed separately?

Reply:
added conditional to resolve this issue.